### ISSUE TITLE
Stripe creates multiple duplicate payments on order review confirmation

### ISSUE THREAD

**Issue by sigoulder** [tryshoppe/shoppe#325]
When you confirm an order after previously entering your card details it validates as it should, however when you view the order in the admin panel there are usually 5 or 6 duplicate payments. Presumably its something in the `confirm!` method being called in the confirmation method

this is what my orders controller looks like: 

```ruby
  class OrdersController < ApplicationController
    before_filter(only: [:confirmation]) do 
      Shoppe::Stripe.setup
    end
 
  ...

    def confirmation
       if request.post?
         current_order.confirm!
         session[:order_id] = nil
         session[:order_has_discount] = nil # remove discount on order session
         redirect_to root_path, :notice => "Order has been placed successfully!"
       end
  end
  ...
```

Any ideas how to overcome this?

Thanks!

**Comment by deanperry** [comment]
What version of `shoppe` and `shoppe-stripe` are you using?

**Comment by sigoulder** [comment]
Im using:
shoppe (1.1.2)
shoppe-stripe (1.2.2)

**Comment by deanperry** [comment]
Hmm ok. Does it only happen if the user enters card details, goes to confirm, changes their mind and adds something else to the basket to then go through the checkout again?

**Comment by sigoulder** [comment]
No it happens every time - even with one item in the basket. On the table of orders the total for that order  will be say £300 but the payment column will say £1200/1500. When you view that order it lists 4/5 of the same payment.

**Comment by deanperry** [comment]
Hmm, I've not seen this issue before so I'm not sure. What if you go back a
couple of versions of Shoppe to see if it sill occurs?

On Thu, 7 Jan 2016 at 09:27, Si Goulder <notifications@github.com> wrote:

> No it happens every time - even with one item in the basket. On the table
> of orders the total for that order will be say £300 but the payment column
> will say £1200/1500. When you view that order it lists 4/5 of the same
> payment.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tryshoppe/shoppe/issues/325#issuecomment-169607259>.
>
-- 
Dean Perry

**Comment by sigoulder** [comment]
Yeah its odd. I've just checked and it looks like it's been doing it from the beginning. Another thing is the order balance is also different to the `total` and `payment` columns, but still a multiple of `total` so presumably its hitting the stripe API multiple times.

It might be something in my controller logic then?

**Comment by sigoulder** [comment]
I've just created a separate app by following the docs pretty much verbatim and i'm encountering the same issue. Still stripe v 1.12 and stripe-shoppe 1.2.2. Not sure if it matters but I'm calling `Shoppe::Stripe.setup` before the confirmation action. Any ideas on how to get around this?
Thanks

**Comment by sigoulder** [comment]
I've just created a separate app by following the docs pretty much verbatim and i'm encountering the same issue. Still stripe v 1.12 and stripe-shoppe 1.2.2. Not sure if it matters but I'm calling `Shoppe::Stripe.setup` before the confirmation action. Any ideas on how to get around this?
Thanks

**Comment by deanperry** [comment]
Odd. I'll see if I can take a look at this and run into the same issue

On Fri, 8 Jan 2016 at 12:07, Si Goulder <notifications@github.com> wrote:

> I've just created a separate app by following the docs pretty much
> verbatim and i'm encountering the same issue. Still stripe v 1.12 and
> stripe-shoppe 1.2.2. Not sure if it matters but I'm calling
> Shoppe::Stripe.setup before the confirmation action. Any ideas on how to
> get around this?
> Thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tryshoppe/shoppe/issues/325#issuecomment-169981689>.
>
-- 
Dean Perry

**Comment by sigoulder** [comment]
Great, thank you!

**Comment by sigoulder** [comment]
Hi, just wondering if you managed to look into this?

**Comment by deanperry** [comment]
Hey, not yet as I've been fairly busy but I'll see if I get some time tonight

**Comment by sigoulder** [comment]
No problem. Many thanks

**Comment by orendon** [comment]
@deanperry @sigoulder wondering if you guys managed to solve this already?

**Comment by sigoulder** [comment]
@orendon The problem arose for me because I was declaring `Shoppe::Stripe.setup` before the order confirmation action. I did this originally as a workaround as I was getting issues with payments being absent from orders being made. It's seemed to resolve itself now though - my versions are still the same as above (stripe v 1.12/stripe-shoppe 1.2.2).

**Comment by lukeheuer** [comment]
> I did this originally as a workaround as I was getting issues with payments being absent from orders being made.

@sigoulder was this happening in production or development mode? There is something causing payments to silently skip being authorized it seems. There's issue #345 open for it.

**Comment by roamingceo** [comment]
@deanpcmad I know this is an old thread, but have you come up with a solution for this?  I'm bouncing back and forth between the issues described here and in #345 , but haven't found a solution for either.

**Comment by deanpcmad** [comment]
I'm afraid not as I no longer use or work on Shoppe

**Comment by roamingceo** [comment]
@deanpcmad Ok.  Thanks for responding!
